how-to/how-to-set-prometheus-in-k8s.md

```diff
@@ -0,0 +1,59 @@
+# How to monitor Kata Containers in K8s clusters
+
+
+This document will describe how to run `kata-magent` in a K8s cluster, and using Prometheus's K8s service discovery to scrape metrics from `kata-agent`.
+
+
+- [Introduction](#introduction)
+- [Pre-requisites](#pre-requisites)
+- [Configure Prometheus](#configure-prometheus)
+- [Configure `kata-magent`](#configure-kata-magent)
+
+> **Warning**: This how-to is only for evaluation purpose, you **SHOULD NOT** running it in production using this configurations.
+
+## Introduction
+
+If you are running Kata containers in a K8s cluster, the best way to run `kata-magent` is using K8s native `DaemonSet`, `kata-magent` will run on desired K8s nodes without other operations when new nodes joined the cluster.
+
+Prometheus also support a K8s service discovery that can find scrape targets dynamically without explicitly setting `kata-magent`'s metric endpoints.
+
+## Pre-requisites
+
+You must have a running K8s cluster first. If not, [install a K8s cluster](https://kubernetes.io/docs/setup/) first.
+
+Also you should ensure that `kubectl` working correctly.
+
+## Configure Prometheus
+
+Running Prometheus is easy.
+
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kata-containers/documentation/master/how-to/data/prometheus.yml
+```
+
+This will create a new namespace named `prometheus`, and create some resources for Prometheus:
+
+* `ClusterRole`, `ServiceAccount`, `ClusterRoleBinding` to let Prometheus to access K8s API server.
+* `ConfigMap` that contains minimum configurations to let Prometheus run K8s service discovery.
+* `Deployment` that run Prometheus in `Pod`.
+* `Service` with `type` of `NodePort`(`30909` in this how to), that we can access Prometheus through `<hostIP>:30909`. In production environment, this `type` may be `LoadBalancer` or `Ingress` resource.
+
+
+After the Prometheus server has running, open `hostIP:NodePort` in browser to verify Prometheus is working correctly.
+
+## Configure `kata-magent`
+
+As Prometheus, running `kata-magent` need only one command too.
+
+```
+$ kubectl apply -f https://raw.githubusercontent.com/kata-containers/documentation/master/how-to/data/kata-magent-daementset.yml
+```
+
+This will create a new namespace `kata-system` and a deployment in it.
```

Comment:
It is actually a daemonset, right?

As an aside from this doc PR: do you think we should move kata-deploy into this namespace as well (and future operator)?

Reply:
Yes, it's a daemonset.

Vendor product should run in its own namespace, and Kata containers workload also should run in the same namespace.

Reply:
OK, can you s/deployment/daemonset ?

Reply:
Done.